sol-anchor/idl.go

```diff
@@ -0,0 +1,612 @@
+	Name string  `json:"name"`
+	Args []Field `json:"args"`
+}
+
+func (i *Instruction) PascalCaseName() string {
+	return textcase.PascalCase(i.Name)
+}
+
+func (i *Instruction) SnakeCaseName() string {
+	return toSnakeCase(i.Name, true)
+}
+
+// --- FIELDS
+
+type Field struct {
+	Name string    `json:"name"`
+	Type FieldType `json:"type"`
+	//Index bool      `json:"index"`
+}
+
+func (f *Field) SnakeCaseName() string {
+	return toSnakeCase(f.Name, true)
+}
+
+func (f *Field) SnakeCaseNameWithoutInitialUnderscore() string {
+	return toSnakeCase(f.Name, false)
+}
+
+// FIELD TYPE
+
+type FieldTypeArray struct {
+	Type   string
+	Length string
+}
+
+type FieldTypeVec struct {
+	Type      string
+	IsDefined bool
+}
+
+type FieldTypeOption struct {
+	Simple  *string
+	Defined *string
+	Vec     *FieldTypeVec
+	Array   *FieldTypeArray
+}
+
+func (f *FieldTypeOption) IsSimple() bool {
+	return f.Simple != nil
+}
+
+func (f *FieldTypeOption) IsDefined() bool {
+	return f.Defined != nil
+}
+
+func (f *FieldTypeOption) IsVec() bool {
+	return f.Vec != nil
+}
+
+func (f *FieldTypeOption) IsArray() bool {
+	return f.Array != nil
+}
+
+type FieldType struct {
+	Simple  string
+	Defined string
+	Array   *FieldTypeArray
+	Vec     *FieldTypeVec
+	Option  *FieldTypeOption
+}
+
+func (t *FieldType) IsSimple() bool {
+	return t.Simple != ""
+}
+
+func (t *FieldType) IsSimplePubKey() bool {
+	return t.Simple == "publicKey"
+}
+
+func (t *FieldType) IsDefined() bool {
+	return t.Defined != ""
+}
+
+func (t *FieldType) IsArray() bool {
+	return t.Array != nil
+}
+
+func (t *FieldType) IsVec() bool {
+	return t.Vec != nil
+}
+
+func (t *FieldType) IsOption() bool {
+	return t.Option != nil
+}
+
+func (t *FieldType) Resolve() string {
+	if t.IsSimplePubKey() {
+		return "PubKey"
+	}
+
+	if t.IsSimple() {
+		return t.Simple
+	}
+
+	if t.IsDefined() {
+		return t.Defined
+	}
+
+	if t.IsArray() {
+		return t.Array.Type
+	}
+
+	if t.IsVec() {
+		return t.Vec.Type
+	}
+
+	if t.IsOption() {
+		if t.Option.IsSimple() {
+			return *t.Option.Simple
+		}
+
+		if t.Option.IsDefined() {
+			return *t.Option.Defined
+		}
+
+		if t.Option.IsVec() {
+			return t.Option.Vec.Type
+		}
+
+		if t.Option.IsArray() {
+			return t.Array.Type
+		}
+	}
+
+	return ""
+}
+
+func (t *FieldType) ResolveProtobufType() string {
+	return ToProtobufType(t.Resolve())
+}
+
+func PrintSimple(simpleType string, fieldName string, variableName string, isOptional bool) string {
+	fieldNameSnakeCase := toSnakeCase(fieldName, true)
+	fieldNameSnakeCaseWithoutInitialUnderscore := toSnakeCase(fieldName, false)
+
+	// If cast is needed
+	cast := CastInRustIfNeeded(simpleType)
+	if cast != "" {
+		if isOptional {
+			return fmt.Sprintf("%s: Some(%s.%s as %s),", fieldNameSnakeCaseWithoutInitialUnderscore, variableName, fieldNameSnakeCase, cast)
+		}
+		return fmt.Sprintf("%s: %s.%s as %s,", fieldNameSnakeCaseWithoutInitialUnderscore, variableName, fieldNameSnakeCase, cast)
+	}
+
+	// If cast is NOT needed
```

Comment:
```suggestion
	// If cast is needed
```

You can write an example of what the function is doing above it. But those comments are not meaningful